src/atlas/functionspace/detail/StructuredColumns_setup.cc

```diff
@@ -588,13 +588,20 @@ void StructuredColumns::setup(const grid::Distribution& distribution, const ecki
 
         atlas_omp_parallel_for(idx_t n = 0; n < gridpoints.size(); ++n) {
             const GridPoint& gp = gridpoints[n];
-            if (gp.j >= 0 && gp.j < grid_->ny()) {
-                xy(gp.r, XX) = grid_->x(gp.i, gp.j);
-                xy(gp.r, YY) = grid_->y(gp.j);
-            }
-            else {
-                xy(gp.r, XX) = compute_x(gp.i, gp.j);
-                xy(gp.r, YY) = compute_y(gp.j);
+            if (regional) {
+              std::vector<double> lonlatVec(2);
```

Comment:
```suggestion
              std::array<double,2> lonlatVec;
```
This avoids heap allocation within the loop over grid points, and should give very nice speedup.

Reply:
Done.